### PR TITLE
feat(translations): cleaning and adding scripts for translations - working toward automation

### PR DESCRIPTION
**Pull request by rusackas**
### SUMMARY

Ok, so we're working toward solidifying/stabilizing translation efforts. Here's another step in the chain, where we:

* Remove `.json` translation files from the repo and add them to `.gitignore`
* Tweak the `.json` building script so it can be run from a new `npm run build-translations` script
* Build those translations whenever doing `npm run build` or `build-instrumented`
* Remove translation code owners - they shouldn't be needed now that people won't likely edit the `.json` files.
* Removes two `npm` scripts for linting translations. From here on out, you just get what you get, and it'll work :)
* Removing translation lint check on CI, as that's no longer relevant.

Next steps/PRs:
* add a CI task that updates the `.pot` file and `.po` files on any PR that touches `.py` or `.(j|t)s(x)` files. 
* build the translation files on docker builds

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: .github/workflows/superset-translations.yml [hunk 1]
```diff
@@ -29,10 +29,6 @@ jobs:
         uses: ./.github/actions/cached-dependencies
```
Comment (mistercrunch) [inline .github/workflows/superset-translations.yml:29]: I think this file should be `frontend-translation` as it only does that (doesn't do backend)

**Review by mistercrunch — APPROVED** [review]
LGTM

**Comment by rusackas** [comment]
Wow... of all things, the DB tests are failing spectacularly. Not sure what to make of this yet.

File: superset-frontend/package.json [hunk 2]
```diff
@@ -38,12 +38,14 @@
   ],
   "scripts": {
     "_prettier": "prettier './({src,spec,cypress-base,plugins,packages,.storybook}/**/*{.js,.jsx,.ts,.tsx,.css,.less,.scss,.sass}|package.json)'",
-    "build": "cross-env NODE_OPTIONS=--max_old_space_size=8192 NODE_ENV=production BABEL_ENV=\"${BABEL_ENV:=production}\" webpack --mode=production --color",
+    "build": "npm run build-translations && cross-env NODE_OPTIONS=--max_old_space_size=8192 NODE_ENV=production BABEL_ENV=\"${BABEL_ENV:=production}\" webpack --mode=production --color",
```
Comment (mistercrunch) [inline superset-frontend/package.json:41]: I wouldn't assume that everyone wants translations as it slows down build time and makes the build heavier, especially given the unknown/incomplete state of various translations. I'd add this as an optional step for those interested in it.

**Comment by mistercrunch** [comment]
I think everything here is superseeded by https://github.com/apache/superset/pull/28483

**Pull request closed without merging**